$OpenBSD: patch-lib_puppet_type_package_rb,v 1.2 2018/11/12 07:17:36 sebastia Exp $

add new feature flavorable, and switch handling the flavor
as a property instead of a parameter

Index: lib/puppet/type/package.rb
--- lib/puppet/type/package.rb.orig
+++ lib/puppet/type/package.rb
@@ -65,6 +65,7 @@ module Puppet
       provider-specific.",
       :methods => [:package_settings_insync?, :package_settings, :package_settings=]
     feature :virtual_packages, "The provider accepts virtual package names for install and uninstall."
+    feature :flavorable, "The provider accepts package flavors."
 
     ensurable do
       desc <<-EOT
@@ -441,7 +442,7 @@ module Puppet
       newvalues(:true, :false)
     end
 
-    newparam(:flavor) do
+    newproperty(:flavor, :required_features => :flavorable) do
       desc "OpenBSD supports 'flavors', which are further specifications for
         which type of package you want."
     end
